### ISSUE TITLE
How to clear the messages stored in the antenna ?

### ISSUE THREAD

**Issue by ZeDvD** [gotenna/PublicSDK#29]
Hi

Juste receveived my goTenna Mesh and used it with myApp. I built a first app with Gotenna V1, and then upgrade to Mesh.

When debugging my app, I need to stop it and restart again. And when restarting, I receive previously sent messages : I think that the antenna keep in memory old messages, and send them once my app is restarted. Is it right ? And how can I clear the old messages when launching my app ?

**Comment by tcolligan-ap** [comment]
@ZeDvD If the goTenna receives a message while an app is not connected, it will save the message so that is can be pulled at a later date. Then, when the app/sdk re-connects to the goTenna, it will pull all the messages that are stored on the goTenna. Every message is deleted after it is pulled by the sdk. There is currently no easy way to clear the goTenna of all messages through a command in the sdk.

**Comment by ZeDvD** [comment]
Thank you. Do you think that this kind of method can be added to the SDK ? I need to do it when calling      startListening(), to clear all the messages at the beginning of my app.

Two other questions :
- What is the goal of the nudgeQueue command in GTCommandCenter ?
- Where can I find the latest firmware udpdate ?

**Comment by tcolligan-ap** [comment]
@ZeDvD I'll add the feature to our internal tracker, but it is going to be pretty low priority since nobody else has requested that feature, I am not sure if/when we will get around to it.

* You can ignore nudgeQueue, that is not a function you need to be calling.
* The newest sample app has a good example of how to a do a Firmware Update, I would suggest taking a look at that.

**Comment by dferg7** [comment]
Just to comment on this, 

I would also like to see the feature of being able to clear all the messages before connecting. High priority on my end. Currently, I'm using the SDK to send GT messages with data that is plot on a map and if they are old messages they are useless and plot points that I don't need. It becomes more of a hassle to remove these markers that then reside on the map.

**Comment by tcolligan-ap** [comment]
@dferg7 You might be able to implement a workaround in the meantime. If you are using the built in GT messages, they all inherit from `GTBaseMessageData` which has a function `getMessageSentDate()`. 

You could potentially use that date as a filter on incoming messages and discard old ones. Of course this isn't fool proof though, since that date is the date the message was sent according to the clock on the sender phone, so your mileage may vary is the sender has played around with the clock on their device.

**Comment by dferg7** [comment]
@tcolligan-ap good idea, thank you! I do have a question unrelated to this post but I don't really think I need to open a whole "issue" for it. After connecting with the Mesh goTenna and calling the function: GTCommandCenter.getInstance().sendSetGeoRegion(Place.NORTH_AMERICA, responseListener, errorListener);
what would be the correct goTenna error to check for? There really isn't any errors directly relating to setting the region.

**Comment by tcolligan-ap** [comment]
@dferg7 You do not really need to check for any specific error. If you get a Positive response for that command you are good to go, otherwise, there was a problem setting the region and you should probably try again. The goTenna device does cache the last region that was set so if the command does not go through for some reason later on, it is not the end of the world. It will still be set to whatever Region it was set to previously.

**Comment by ZeDvD** [comment]
A new request from myself for this feature : here is my issue. 

My app starts by creating a group. But if there are messages stored in the antenna, they are first read before the group request, this may take 30 sec. The group creation process fails because to many messages are currently read. So I must first wait that all the messages have been read before being able to send my group creation request.

Is it possible to know how many messages are stored in the Antenna ? 

And I really think that a "clear message" feature would be useful to avoid this kind of issue.
